rs/sns/swap/src/swap.rs

```diff
@@ -1160,6 +1162,30 @@ impl Swap {
         // Subtraction safe because of the preceding if-statement.
         let max_increment_e8s = self.available_direct_participation_e8s();
 
+        // Check that the maximum number of participants has not been reached yet.
+        {
+            let num_direct_participants = self.buyers.len() as u64;
+            let num_sns_neurons_per_basket = params
+                .neuron_basket_construction_parameters
+                .as_ref()
+                .expect("neuron_basket_construction_parameters must be specified")
+                .count;
+            if (num_direct_participants + 1) * num_sns_neurons_per_basket
+                > MAX_NEURONS_FOR_DIRECT_PARTICIPANTS
+            {
+                return Err(format!(
```

Comment:
Is there a test somewhere?

Reply:
Added `test_refresh_buyer_tokens_no_sns_neuron_baskets_available` in rs/sns/swap/tests/swap.rs